UnleashClient/__init__.py

```diff
@@ -186,14 +196,44 @@ def is_enabled(self,
             except Exception as excep:
                 LOGGER.warning("Returning default value for feature: %s", feature_name)
                 LOGGER.warning("Error checking feature flag: %s", excep)
-
-                if fallback_function:
-                    fallback_value = default_value or fallback_function(feature_name, context)
-                else:
-                    fallback_value = default_value
-
-                return fallback_value
+                return self._get_fallback_value(default_value, fallback_function, feature_name, context)
         else:
             LOGGER.warning("Returning default value for feature: %s", feature_name)
             LOGGER.warning("Attempted to get feature_flag %s, but client wasn't initialized!", feature_name)
-            return default_value
+            return self._get_fallback_value(default_value, fallback_function, feature_name, context)
+
+
+    # pylint: disable=broad-except
+    def select_variant(self,
+                       feature_name: str,
+                       context: dict = {},
+                       default_value: bool = False,
+                       fallback_function: Callable = None) -> dict:
```

Comment:
I don't think `default_value` and `fallback_function` makes sense here. The intention for variants is that the user should be allowed to specify a `default_variant` type in her code in the case of the toggle not having a variant to return for this specific toggle and context. Being allowed to define the default/fallback variant can make sense in many situations where the `disabled` variant is not sufficient (e.g. you require some payload).

Reply:
The `default_value` / `fallback_function` here is used for determining whether or not the feature flag is enabled (it's basically passed through to the `is_enabled()` method.

I figure it'd be nice to have a similar interface, though I can also see a little confusion on what exactly is happening.  I could also just remove both and the feature flag portion would just default to False.

Let me know what you'd prefer! :)

Reply:
I think the ability to define the returned `variant` is more appropriate than a default_value.

I have not thought sufficient much about the need to for a "fallback" function. In any case, such a function should return a variant and not a boolean at least.

Reply:
As discussed, removing default/fallback functionality (for now!).  =)